plasmapy/diagnostics/thomson.py

```diff
@@ -111,12 +129,39 @@ def spectral_density(
 
     electron_vel : `~astropy.units.Quantity`, shape (Ne, 3), optional
         Velocity of each electron component in the rest frame. (convertible to m/s)
+        If set, overrides electron_vdir and electron_speed.
         Defaults to a stationary plasma [0, 0, 0] m/s.
 
+    electron_vdir : np.ndarray, shape (Ne,3), optional
+        Unit vectors describing the velocity of each electron population.
+        Unit vectors will be normalized, but must not contain all zero elements.
+        Setting electron_vel overrides this keyword.
+
+    electron_speed : `~astropy.units.Quantity`, shape (Ne), optional
+        A scalar speed for each electron population. Must be used along with
+        electron_vdir. The electron_vel is calculated from these keywords as
+
+        electron_vel = electron_speed * electron_vdir
+
+        Setting electron_vel overrides this keyword.
```

Comment:
Why are these defined as two separate arguments?  I would suspect you could calculate `electron_vdir` directly from  `electron_vel`.